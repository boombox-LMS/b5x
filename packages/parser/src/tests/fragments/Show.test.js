const {
  exampleShowMarkup,
} = require("../../../dist/componentBuilders/fragments/Show");
const { RawFragmentSchema, DraftConditionsListSchema } = require("@b5x/types");

describe("The Show fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(exampleShowMarkup);
  const rawFragment = fragment.packageForApi();

  test("The parsed show fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });
  test("The parsed show fragment is a valid RawFragment", () => {
    // Because the topic is responsible for revising draft conditions
    // to final conditions, an individually built show fragment
    // will still have a draft conditions list instead of a finalized conditions list.
    // TODO: This means that the Show fragment does not return the RawFragment
    // type that it is said to return via the abstract fragment class ... revise?
    // Pass the packageForApi() function a set of uris to fill in with?
    // Some other approach?
    const RawFragmentWithDraftConditionsSchema = RawFragmentSchema.omit({
      displayConditions: true,
    }).extend({
      displayConditions: DraftConditionsListSchema,
    });
    const validator = () => {
      RawFragmentWithDraftConditionsSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });
});
