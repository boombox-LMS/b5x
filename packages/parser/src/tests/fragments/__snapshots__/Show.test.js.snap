// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Show fragment builds as expected The parsed show fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [
    "test-fragment-uri",
  ],
  "contentType": "Show",
  "contents": "",
  "data": Object {},
  "dependencyUris": Array [],
  "digest": "89b8bb9e8d98c83a9e19d98ff9f0f3db",
  "displayConditions": Array [
    Object {
      "data": Object {
        "alias": "favorite-color",
        "operator": "is",
        "targetValue": "hot-pink",
      },
      "type": "condition",
    },
  ],
  "documentUri": "mock-document-uri",
  "isRequired": false,
  "isStateful": false,
  "uri": "test-fragment-uri",
}
`;
